lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php

```diff
@@ -62,17 +62,33 @@ public function setReferenceRepository(ReferenceRepository $referenceRepository)
     }
 
     /** @inheritDoc */
-    public function execute(array $fixtures, $append = false)
+    public function execute(array $fixtures, $append = false /* bool $singleTransaction = false */)
     {
-        $executor = $this;
-        $this->em->transactional(static function (EntityManagerInterface $em) use ($executor, $fixtures, $append) {
+        $singleTransaction = func_get_args()[2] ?? true;
+
+        if ($singleTransaction) {
+            $executor = $this;
+            $this->em->transactional(static function (EntityManagerInterface $em) use ($executor, $fixtures, $append) {
+                if ($append === false) {
+                    $executor->purge();
+                }
+
+                foreach ($fixtures as $fixture) {
+                    $executor->load($em, $fixture);
+                }
+            });
+        } else {
             if ($append === false) {
-                $executor->purge();
+                $this->em->beginTransaction();
+                $this->purge();
+                $this->em->commit();
             }
 
             foreach ($fixtures as $fixture) {
-                $executor->load($em, $fixture);
+                $this->em->beginTransaction();
```

Comment:
The usage of `EntityManager::transactional()` was adding an implicit `flush` that you removed. That's a BC break.

Reply:
I didn't change this part but I'll use the closure style everywhere

Reply:
well, you *did* change that part, by switching from `EntityManager::transactional()` to using `beginTransaction`

Reply:
`$singleTransaction` would be true by default and I kept the same behavior there to avoid BC, I used `beginTransaction` for the new part only (the else), maybe I'm missing your point but I just indented the previous code to have it under the if, the git diff doesn't help.